src/System/Hooks/usePrefetchRoute.tsx

```diff
@@ -0,0 +1,82 @@
+import { RouteProps } from "found"
+import { MatchResult } from "path-to-regexp"
+import { fetchQuery } from "react-relay"
+import { OperationType } from "relay-runtime"
+import { useRouter } from "System/Hooks/useRouter"
+import { useSystemContext } from "System/Hooks/useSystemContext"
+import { findRoutesByPath } from "System/Router/Utils/routeUtils"
+
+interface UsePrefetchRouteProps {
+  path: string
+}
+
+export interface RouteMatch {
+  route: RouteProps
+  match: MatchResult<any>
+}
+
+export const usePrefetchRoute = (
+  props: UsePrefetchRouteProps
+): { prefetch: () => any } => {
+  const { relayEnvironment } = useSystemContext()
+
+  const { match } = useRouter()
+
+  const prefetch = () => {
+    const routes = findRoutesByPath({ path: props.path })
+
+    routes.forEach(foundRoute => {
```

Comment:
This is somewhat eager just in the sense that in our router we sometimes have multiple matchable routes for a given path. 

A good example of this [is here](https://github.com/artsy/force/blob/damassi/feat/prefetch-on-hover/src/Apps/Artist/artistRoutes.tsx#L132), in the `artistRoutes`. We have the parent route (`/artist/:artistID`) which contains `children` routes, of which one child route path (the default subroute) is the works for sale route, keyed by `path: ""`. So this automatically matches when navigating to `/artist/foo`. There's a few places in our router like this, so lets be safe and fetch all related stuff. This might lead to less matches, however.